src/ChangeSet.ts

```diff
@@ -84,7 +96,7 @@ export class ChangeSet {
         const result = this.matchAndAddToRootChange(rootNodes, c)
         if (result) {
           const { index, root } = result
-          rootNodes[index] = root
+          rootNodes[index][0] = root
```

Comment:
please make a note somewhere that item [0] is considered a main change

Reply:
updated